src/ast.ts

```diff
@@ -262,13 +264,12 @@ function augmentFieldNodeTree(
       case Kind.FRAGMENT_SPREAD: {
         const fragment = compilationContext.fragments[selection.name.value];
         for (const subSelection of fragment.selectionSet.selections) {
-          handle(parentFieldNode, subSelection);
+          handle(parentFieldNode, subSelection, true);
```

Comment:
is this change not applicable for InlineFragments?

Reply:
No, in fact our workaround until now was rewrite queries to use inline fragments, following same example if you make next query is working as expected in main branch:
```
query TEST(
    $includeOnly: Boolean!
){
    post(id: "post:1") {
        title
        Author1: author @skip(if: $includeOnly) {
            ...on User {
              name
          }
        }
        Author2: author @include(if: $includeOnly) {
            ...on User {
                name
            }
        }
    }
}
```

And for the field name on first inline fragment you get this correct computed condition:
```
(__context.variables["includeOnly"] === false) && (__context.variables["includeOnly"] === false) 
```

And second gets also the correct one:
```
(__context.variables["includeOnly"] === true) && (__context.variables["includeOnly"] === true) 
```

Reply:
We should add tests to answer this question.

Reply:
i update the tests...